File: 3.x.x/crates-utopia/storage/src/device_type.rs
/// Storage backend kind.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum DeviceType {
    Local,
    #[cfg(feature = "s3")]
    S3,
    #[cfg(feature = "s3")]
    AwsS3,
    #[cfg(feature = "s3")]
    DoSpaces,
    #[cfg(feature = "s3")]
    Wasabi,
    #[cfg(feature = "s3")]
    Backblaze,
    #[cfg(feature = "s3")]
    Linode,
}

impl DeviceType {
    pub const fn as_str(self) -> &'static str {
        match self {
            Self::Local => "local",
            #[cfg(feature = "s3")]
            Self::S3 => "s3",
            #[cfg(feature = "s3")]
            Self::AwsS3 => "awss3",
            #[cfg(feature = "s3")]
            Self::DoSpaces => "dospaces",
            #[cfg(feature = "s3")]
            Self::Wasabi => "wasabi",
            #[cfg(feature = "s3")]
            Self::Backblaze => "backblaze",
            #[cfg(feature = "s3")]
            Self::Linode => "linode",
        }
    }
}

File: packages/storage/src/DeviceType.php
<?php

declare(strict_types=1);

namespace Utopia\Storage;

enum DeviceType: string
{
    case Local = 'local';

    case S3 = 's3';

    case AwsS3 = 'awss3';

    case DoSpaces = 'dospaces';

    case Wasabi = 'wasabi';

    case Backblaze = 'backblaze';

    case Linode = 'linode';
}
